be/src/vec/core/block.cpp

```diff
@@ -678,11 +682,14 @@ Block Block::copy_block(const std::vector<int>& column_offset) const {
     return columns_with_type_and_name;
 }
 
-void Block::append_block_by_selector(MutableColumns& columns,
-                                     const IColumn::Selector& selector) const {
-    DCHECK(data.size() == columns.size());
+void Block::append_block_by_selector(MutableBlock* dst, const IColumn::Selector& selector) const {
+    if (dst->get_block_type() == BlockType::DYNAMIC) {
```

Comment:
It may be more efficient to add UNLIKELY since most tables are without dynamic schema.